typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -1,89 +1,727 @@
+
+import { attachAndConnectContracts } from '../contracts/contracts.js';
+import { HyperlaneAddresses, HyperlaneContracts } from '../contracts/types.js';
 import {
   HyperlaneModule,
   HyperlaneModuleArgs,
 } from '../core/AbstractHyperlaneModule.js';
 import { HyperlaneDeployer } from '../deploy/HyperlaneDeployer.js';
-import { ProxyFactoryFactories } from '../deploy/contracts.js';
+import {
+  ProxyFactoryFactories,
+  proxyFactoryFactories,
+} from '../deploy/contracts.js';
+import { extractOwnerAddress } from '../deploy/types.js';
 import { MultiProvider } from '../providers/MultiProvider.js';
-import { EthersV5Transaction } from '../providers/ProviderType.js';
-import { ChainNameOrId } from '../types.js';
+import {
+  AnnotatedEthersV5Transaction,
+  createAnnotatedEthersV5Transaction,
+} from '../providers/ProviderType.js';
+import { ChainMap, ChainName, ChainNameOrId } from '../types.js';
+import { findMatchingLogEvents } from '../utils/logUtils.js';
 
-import { EvmIsmCreator } from './EvmIsmCreator.js';
 import { EvmIsmReader } from './EvmIsmReader.js';
-import { IsmConfig } from './types.js';
+import {
+  AggregationIsmConfig,
+  DeployedIsm,
+  DeployedIsmType,
+  IsmConfig,
+  IsmType,
+  MultisigIsmConfig,
+  RoutingIsmConfig,
+  RoutingIsmDelta,
+} from './types.js';
+import { moduleMatchesConfig, routingModuleDelta } from './utils.js';
+
+type ExtraArgs = {
+  deployedIsm: Address;
+  mailbox: Address;
+};
 
 export class EvmIsmModule extends HyperlaneModule<
   ProtocolType.Ethereum,
   IsmConfig,
-  HyperlaneContracts<ProxyFactoryFactories> & {
-    deployedIsm: Address;
-  }
+  HyperlaneAddresses<ProxyFactoryFactories> & ExtraArgs
 > {
-  protected logger = rootLogger.child({ module: 'EvmIsmModule' });
-  protected reader: EvmIsmReader;
-  protected creator: EvmIsmCreator;
+  protected readonly logger = rootLogger.child({ module: 'EvmIsmModule' });
+  protected readonly reader: EvmIsmReader;
+  protected readonly factories: HyperlaneContracts<ProxyFactoryFactories>;
+
+  // Adding these to reduce how often we need to grab from MultiProvider.
+  public readonly chainName: string;
+  // We use domainId here because MultiProvider.getDomainId() will always
+  // return a number, and EVM the domainId and chainId are the same.
+  public readonly domainId: number;
 
   protected constructor(
     protected readonly multiProvider: MultiProvider,
     protected readonly deployer: HyperlaneDeployer<any, any>,
     args: HyperlaneModuleArgs<
       IsmConfig,
-      HyperlaneContracts<ProxyFactoryFactories> & {
-        deployedIsm: Address;
-      }
+      HyperlaneAddresses<ProxyFactoryFactories> & ExtraArgs
     >,
   ) {
     super(args);
+
     this.reader = new EvmIsmReader(multiProvider, args.chain);
-    this.creator = new EvmIsmCreator(deployer, multiProvider, args.addresses);
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { mailbox, deployedIsm, ...addresses } = args.addresses;
```

Comment:
```suggestion
    const { mailbox, deployedIsm, ..._addresses } = args.addresses;
```
does this work? I think it can be unused if you prefix with `_`

Reply:
```suggestion
    const { mailbox: _, deployedIsm: __, ...addresses } = args.addresses;
```
no I'd have to do something like this instead, because `_mailbox` and `_deployedIsm` are not properties of `args.addresses`